### ISSUE TITLE
Parser breaks when trying to parse element with an invalid self closing tag

### ISSUE THREAD

**Issue by straker** [angular-eslint/angular-eslint#1105]
**Description and reproduction of the issue**

When a template has an element that uses an invalid self closing tag, the parser breaks and does not give any information about the entire template. For example, given the following template:

```html
<div>hello</div>
<div />
<div>world</div>
```

The returned object structure does not contain any information

```json
{
  "type": "Program",
  "comments": [],
  "tokens": [],
  "range": [
    0,
    0
  ],
  "loc": {
    "start": {
      "line": 0,
      "column": 0
    },
    "end": {
      "line": 0,
      "column": 0
    }
  },
  "templateNodes": [],
  "value": "<div>hello</div>\n<div/>\n<div>world</div>"
}
```

My expected behavior is that the parser is able to give information about the valid elements and properly handle the invalid self closing tag. How it's handled is different for each parser, but it seems a lot of them add a closing tag after the self closing element to make it a div with no children (htmlparser2, svelte, vue-eslint-parser, etc.)

Snippet: 
https://astexplorer.net/#/gist/bb89902e694482324e3fe0da79802387/d88837b70fae35f9afb1a51888540d449b961ca6 

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/template-parser`  | `12.3.1` |
| `ESLint`                           | `8` |
| `node`                             | `16` |

**Angular version**

```
Angular CLI: 12.2.6
Node: 16.16.0 (Unsupported)
Package Manager: yarn 1.22.18
OS: darwin x64

Angular: undefined
...

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1202.6 (cli-only)
@angular-devkit/core         12.2.6 (cli-only)
@angular-devkit/schematics   12.2.6 (cli-only)
@schematics/angular          12.2.6 (cli-only)
```

- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
Hi @straker,

Parser errors have been correctly propagated for some time, this is multiple major versions behind (as is the version on AST explorer).

Please try again with the latest.

Many thanks 🙏
